src/deprecated.jl

```diff
@@ -1402,4 +1402,7 @@ import Base: haskey
 @deprecate haskey(df::AbstractDataFrame, key::Any) key in names(df)
 
 import Base: empty!
-@deprecate empty!(df::DataFrame) deletecols!(df, 1:ncol(df))
\ No newline at end of file
+@deprecate empty!(df::DataFrame) deletecols!(df, 1:ncol(df))
+
+@deprecate deletecols!(df::DataFrame, inds) select!(df, Not(inds))
+@deprecate deletecols(df::DataFrame, inds; copycols::Bool=true) select(df, Not(inds), copycols=copycols)
```

Comment:
Maybe also add a version without keyword arguments to keep the most common case simple.

Reply:
It is not possible unfortunately as `copycols` is a kwarg so then we would have two declarations with the same signature and one `@deprecate` would get overwritten by the other.

Reply:
Actually you can cheat by using different types for positional arguments (e.g. `AbstractDataFrame`), but maybe that's not worth it. See https://github.com/JuliaLang/julia/issues/9498.

Reply:
Actually it works even when the types are the same:
```julia
julia> f(x::Int; y=1) = 2
f (generic function with 1 method)

julia> f(x::Int) = 1
f (generic function with 1 method)

julia> f(1)
1

julia> f(1, y=2)
2
```
(That's probably one of the most obscure "features" of dispatch.)

Reply:
I have tested a reverse order I have checked when originally replying to you. I have commented on it in https://github.com/JuliaLang/julia/issues/9498. I will add the other definition then.